### DISCUSSION TITLE
Dags not loading in docker environment

### DISCUSSION THREAD

**Discussion by zacqed** [apache/airflow#25535] [General]
### Apache Airflow version

2.3.3 (latest released)

### What happened

We have deployed Airflow docker containers  with mssql backend and are unable to load a new dag automatically on the webserver front end. We need to go ahead and reset the entire DB and reload airflow in order for the dag to start appearing on the front end. The dag is visible when we do docker dags list:

![image](https://user-images.githubusercontent.com/1675591/182875681-07278ef8-8e86-4d06-b8ce-4afbc5d1f1b8.png)

1  This is a running airflow enviornment with pre-loaded dags
2. A new dag in the screenshot above the last file is placed in the dags directory
3. Inside the web container, we can see the dag is being fetched using airflow dags list
4. However on the front end the same is not visible
5. Going through the documentation we could find the below scheduler setting and the same has been added to the docker compose file under environment 
docker compose under environment has the setting:
` AIRFLOW__SCHEDULER__DAG_DIR_LIST_INTERVAL: '60'`

However this is not visible on the front end.

### What you think should happen instead

The new dag should be loaded on the front end browser

### How to reproduce

1  Connect a airflow docker container to mssql db
2 load any dag and run the containers using docker compose 
3 create a new dag and place it the dags directory mounted to the container
4 wait for 5 mins for it appear on the front end

### Operating System

debian

### Deployment

Docker-Compose

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
There are not nearly enough detail here to help you with your issues. If you want someone to help you, you must do much better job in documenting what your problem is.

**Reply by zacqed** [reply]
Hi, I have updated with more information, let me know if you need any further information.

Apologies, intent isn't to file a blind issue.

**Comment by potiuk** [comment]
But  you seem to have some warnings and inconsistencies between configuration of your CLI and the rest. Make sure to follow this:

> Use the same configuration across all the Airflow components. While each component does not require all, some configurations 
need to be same otherwise they would not work as expected. A good example for that is [secret_key](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#config-webserver-secret-key) 
which should be same on the Webserver and Worker to allow Webserver to fetch logs from Worker.

And generally studying what starts in this page https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html (and following from there) is a good way to start debugging your problem.

**Reply by zacqed** [reply]
This is for config update under Airflow 3.0.0 warning

**Reply by potiuk** [reply]
You must have scheduler runnig for new dags to appear. Make sure you have, and make sure your db and all connections are ok. Pointing to the same db. If not - look at the logs of all the components and find any anomalies there (and fix) and yeah . Make sure you fix all the warnings too. Without you looking at the logs and doing some investigation first, there is no-one who could help you with investigating your issue - which is likely some misconfiguration.
